### ISSUE TITLE
Integrate iOS Document Picker to open/add a database

### ISSUE THREAD

**Issue by dsteinkopf** [MiniKeePass/MiniKeePass#388]
see https://developer.apple.com/library/ios/documentation/FileManagement/Conceptual/DocumentPickerProgrammingGuide/Introduction/Introduction.html

I think this would be a great step forward. In my use case I would like to use it to open/save a keepass db stored on a owncloud. Currently I have to push all updates in my keepass db to MiniKeePass manually.

Somewhere in the dialog that opens when hitting the "+" would be the right place in my opinion.

**Comment by dsteinkopf** [comment]
PS. Maybe I could do the implementation if this is desired.

**Comment by jarush** [comment]
We actually have a documentpicker branch where we implemented the basics.  If I recall, there were some issues with how the document picker interacted with the lock screen.

I think it would be a nice feature for MiniKeePass, and we'd welcome a pull request if you wanted to work on it.  You could take a look where we left off on that branch (it's probably out of sync with master), or just starting from scratch since I recall it being fairly straight forward to implement.

**Comment by dsteinkopf** [comment]
Ok. Fine. I already played with MiniKeePass and DokumentPicker a bit...

Is it correct, that your document picker is mainly just the commit 9110b75 ? (It's about as far as my experiments were develeped :-) )
And is it correct you just undid these changes in master later? (I tried to merge master to documentpicker-branch and all DocumentPicker changes were gone.)

And I would be happy to hear from you what kind of problems you had with the lock screen. So I don't have to do the same trial and error. Currently I can't see any problems on my device (also using your implementation from documentpicker-branch)

**Comment by jarush** [comment]
You're right, it was a commit we eventually backed out of master, I had forgotten about that. There might not be anything useful in that branch.

The problem we had was when you had another app's doc picker UI on the screen (like Dropbox, Google Drive, OneDrive, etc), then switched apps so MiniKeePass' PIN lockscreen would appear. I don't recall the specifics, but the lockscreen/PIN view would not behave correctly. If that's no longer an issue, it should be pretty easy to integrate.
